connection.go

```diff
@@ -554,9 +554,23 @@ func (sc *snowflakeConn) FetchResult(ctx context.Context, qid string) (driver.Ro
 // go sql library but is exported to clients who can make use of this
 // capability explicitly.
 func (sc *snowflakeConn) WaitForQueryCompletion(ctx context.Context, qid string) error {
+	// if the query is already done, we dont need to wait for it 
+	complete, err := sc.checkIfComplete(ctx, qid) 
+	if complete {
+		return err
+	}
 	return sc.blockOnQueryCompletion(ctx, qid)
 }
 
+func (sc *snowflakeConn) checkIfComplete(ctx context.Context, qid string) (bool, error) {
+	_, err := sc.checkQueryStatus(ctx, qid)
+	if err == nil || err.(*SnowflakeError).Number != ErrQueryIsRunning {
```

Comment:
Will `err` ever be a non-nil error that's not `*SnowflakeError`?

Reply:
Yeah it can be `ErrQueryReportedError` if the requested query was terminated or aborted
and GS returned an error status included in query. SFQueryFailedWithError, or if there is a connection or permissions error, it would return `ErrQueryStatus` and the associated permission/connection error